### ISSUE TITLE
Serenity/JS 3.0 - Support to locate the not unique parent element by `Text.ofAll` childs

### ISSUE THREAD

**Issue by viper3400** [serenity-js/serenity-js#1084]
Based on the discussion I had today with @jan-molak at https://gitter.im/serenity-js/Contributors?at=61cd86345ee4df335abdd9fb I'd like to raise this request for Serenity/JS 3.

Imagine the follwing HTML code:

```html
  <div class="container">
        <h1>First shopping list</h1>
        <ul>
            <li>coffee</li>
            <li>biscuits</li>
            <li>milk</li>
        </ul>
    </div>
    <div class="container">
        <h1>Second shopping list</h1>
        <ul>
            <li>coffee</li>
            <li>cake</li>
        </ul>
    </div>
    <div class="container">
        <h1>Third shopping list</h1>
        <ul>
            <li>apples</li>
            <li>oranges</li>
        </ul>
    </div>
```

This was possibele with Serenity/JS 2 and `Target.all`

For Serenity/JS 3 the `PageElements` would look like:

```ts
class Elements {
	static containers = () =>
	PageElements.located(By.css('div.container')).describedAs('container');

	static heading = () =>
		PageElement.located(By.css('h1')).describedAs('heading');

	static items = () =>
		PageElements.located(By.css('ul li')).describedAs('shopping list items');
}
```

And this test shall be executable without errors:

```ts
it('gets the second shopping list', () =>
	actorCalled('Elle').attemptsTo(
		Navigate.to('/screenplay/models/page-elements/containers_with_labels.html'),
		Ensure.that(
			Text.of(
				Elements.heading().of(
					Elements.containers()
						.where(Text.ofAll(Elements.items()), contain('coffee'))
						.last()
				)
			),
			equals('Second shopping list')
		),
	));
```

One can also think of something like this:

```ts	
it('gets the first shopping list', () =>
	actorCalled('Elle').attemptsTo(
		Navigate.to('/screenplay/models/page-elements/containers_with_labels.html'),
		Ensure.that(
			Text.of(
				Elements.heading().of(
					Elements.containers()
						.where(Text.ofAll(Elements.items()), contain('coffee'))
						.where(Text.ofAll(Elements.items()), contain('milk'))
						.first()
				)
			),
			equals('Second shopping list')
		),
	));
```

As I mentioned in the chat, I don't consider this test as blocking for my test suite, but I still have to check this. I will inform, if the priority change.

**Comment by jan-molak** [comment]
Could you please see my https://github.com/serenity-js/serenity-js/issues/1099#issuecomment-1023804036 and let me know if the solution introduced in 3.0.0-rc.7 addresses this case too?

**Comment by viper3400** [comment]
I consider this as fixed / implemented and close the issue (tested with 3.0.0.-rc.12)
